rts/Sim/Units/CommandAI/MobileCAI.cpp

```diff
@@ -1901,25 +1915,10 @@ void CMobileCAI::UnloadLand(Command& c)
 
 	SetGoal(wantedPos, owner->pos);
 
-	if (c.GetNumParams() < 4) {
-		// unload the first transportee
-		transportee = transportees[0].unit;
-	} else {
-		const int unitID = c.GetParam(3);
-
-		// unload a specific transportee
-		for (const CUnit::TransportedUnit& tu: transportees) {
-			CUnit* carried = tu.unit;
-
-			if (unitID == carried->id) {
-				transportee = carried;
-				break;
-			}
-		}
-		if (transportee == nullptr) {
-			StopMoveAndFinishCommand();
-			return;
-		}
+	const auto transportee = GetTransporteeFromUnloadOrder(c, transportees);
+	if (transportee == nullptr) {
```

Comment:
Maybe:
```
if (const auto* transportee = GetTransporteeFromUnloadOrder(c, transportees); transportee) {
```
?

Reply:
`transportee` is used a lot afterwards though (from line 1928/1927 until 1972/1971), its scope is the whole rest of the function and not limited just to this check.